### DISCUSSION TITLE
VSCode Debugging Webpack + Vue SFC + Script Setup + TypeScript

### DISCUSSION THREAD

**Discussion by hinogi** [quasarframework/quasar#12799] [General - Components / Directives / etc]
I think this is a rather complex topic because the webpack pipeline will auto import prerequisites for Quasar to run, also transpiles vue to html/css/js, may also inject compiler macros for script setup and/or polyfills for target ES version and on top, also has the HOT stuff from webpack devserver in the Rucksack.

There seem to be sourcemaps for all of this steps each, but none that goes the complete way which renders the breakpoints for your favorite browser debugger adapters in VSCode unable to find the proper codeline for the breakpoint.

If anyone with in-depth knowledge of the webpack chain could maybe add some insights to this on how to maybe be able to pull this off that would be very welcome.

**Comment by pdanpdan** [comment]
Maybe this can help: https://www.mistergoodcat.com/post/the-joy-that-is-source-maps-with-vuejs-and-typescript

**Reply by hinogi** [reply]
Cannot make it work with this instructions, may also because of different webpack versions

**Reply by rdoepke** [reply]
@pdanpdan Thinking about this for several days now, I really do not think that providing links that might or might not help in debugging the debugger is sufficient here.

Allowing debugging with typescript (and SFC script setup) is something that must work out of the box or with well documented settings). I think it is a fundamental feature in professional development. Using SFC with script setup is no weird edge case, where it does not work for some users, but the proposed way to use vue and quasar - with a fresh project. This together with misleading official documentation suggesting that it is easy to get it working (https://quasar.dev/start/vs-code-configuration) - makes it even worse. It seems to work with javascript - but not with typescript. To my understanding, typescript isn't just experimentally supported with vue/quasar.

There was an issue opened in July (see https://github.com/quasarframework/quasar/issues/9614) which was closed stating that it is not a Quasar issue. Blog posts like this one https://ckhconsulting.com/vue-typescript-debuggers/, suggest that it could be possible to get it running - but no one seems to have tackled it down end to end (therefore, see comments in this blog post).

**Reply by pdanpdan** [reply]
I don't disagree. It's just that I don't use ts for front end, so searching for other people experience was what I could do :)

**Reply by hinogi** [reply]
I think it has actually not so much to do with Quasar but with Webpack and how SFCs are deconstructed. The thing is, you have to have 3 separate files for the 3 different parts of a vue file. Since you can have different language settings for all of them (pug, ts, scss for example) so all the parts might need different transpilation.
Also, it might be the case that you have parts that will then inject html/css with js, so it is not that easy to distinguish those in the output from the script only part.
The trick is I guess connect the correct souremaps that are generated on the way and have only the actual script part be reflected in an identifiable file additionally to the hot file that will be generated by the webpack dev server for on-the-fly injection/evaluation.
But therefore you need to identify, where the source map starts to diverge, obviously the dev server does not output sourcemaps to the local filesystem.
Eventually, you might have to test this with other build tools that may build upon tools like rollup, I guess vite would fall into this, and see if they have the same problems.

**Reply by hinogi** [reply]
My hope is, that with the already existing webpack code/dependency injection happening in the current build workflow, and with the help of the blog article above there might be a way to make it happen by identifying and filtering the relevant resulting script files somehow, or have a convention where they appear.

**Reply by jaybo** [reply]
Does switching to Vite change anything regarding debugging and TypeScript?
